service/controller/key/key.go

```diff
@@ -347,6 +348,7 @@ func KubeletLabelsNodePool(getter LabelsGetter) string {
 
 	labels = ensureLabel(labels, label.Provider, "azure")
 	labels = ensureLabel(labels, label.OperatorVersion, OperatorVersion(getter))
+	labels = ensureLabel(labels, apiextensionslabels.MachinePool, MachinePoolID(getter))
```

Comment:
This is the important change. We add a new label to the `Node` object in k8s so that it contains the node pool id. This will allow customers to schedule different workloads on different node pools using the label.